python_modules/dagster-graphql/dagster_graphql_tests/graphql/repo.py

```diff
@@ -2166,6 +2173,12 @@ def define_asset_checks():
 
 
 def _targets_asset_job(instigator: Union[ScheduleDefinition, SensorDefinition]) -> bool:
+    if isinstance(instigator, SensorDefinition) and instigator.sensor_type in (
+        # these contain asset selections, which are invalid with the dict repo
+        SensorType.AUTOMATION,
+        SensorType.AUTO_MATERIALIZE,
+    ):
+        return True
```

Comment:
I'm not folllowing this comment and logic

Reply:
I'll update the comment, the basic idea is that the method of constructing a repository with a dictionary is very old and does not support "unresolved asset jobs".

By that same token, it cannot support AutomationConditionSensorDefinitions (or any type of sensor/schedule that targets an asset selection).